File: HOT100/HOT100p1/5.普通数组/56.合并区间.cpp
/*
 * @lc app=leetcode.cn id=56 lang=cpp
 *
 * [56] 合并区间
 */
#include <bits/stdc++.h>
using namespace std;

// @lc code=start
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        vector<vector<int>> res;
        ranges::sort(intervals);
        res.emplace_back(intervals[0]);
        for (int i = 1; i < intervals.size(); i ++ )
        {
            if (intervals[i][0] <= res.back()[1])
            {
                res.back()[1] = max(res.back()[1], intervals[i][1]);
            }
            else
            {
                res.emplace_back(intervals[i]);
            }
        }
        return res;
    }
};
// @lc code=end

File: HOT100/HOT100p4/56.合并区间.py
#
# @lc app=leetcode.cn id=56 lang=python3
#
# [56] 合并区间
#
from mytools import *
# @lc code=start
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        res = []
        intervals.sort()
        for s, e in intervals:
            if res and s <= res[-1][1]:
                if e > res[-1][1]: res[-1][1] = e
            else: res.append([s, e])
        return res
# @lc code=end

